File: API/QuestGiver/Models/Send/VoteDTO.cs
using QuestGiver.Data.Common;
using QuestGiver.Data.Models;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace QuestGiver.Models.Send
{
    public class VoteDTO
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        [StringLength(200)]
        public string Description { get; set; }

        [Required]
        public VoteType Discriminator { get; set; }

        public DateTime DateCreated { get; set; } = DateTime.UtcNow;

        [Required]
        [ForeignKey(nameof(Quest))]
        public Guid QuestId { get; set; }

        [NotMapped]
        public bool? Decision { get; set; }

        public virtual ICollection<UserVoteDTO> UserVotes { get; set; } = new List<UserVoteDTO>();
    }
}

File: Frontend/src/types/Receive/VoteDTO.ts
import type { VoteType } from "../VoteType";
import type { UserVoteDTO } from "./UserVoteDTO";

export type VoteDTO = {
    id: string,
    description: string,
    discriminator: VoteType,
    dateCreated: Date,
    questId: string,
    decision?: boolean,
    userVotes: UserVoteDTO[]
}